tools/uwuw_preproc

```diff
@@ -461,27 +502,34 @@ Function that writes tally objects to hdf5 file
 tally_list: list of volume id tally pairs of PyNE Material Objects
 filename: filename to write the objects to
 """
-def write_tally_h5m(tally_list,filename):
+
+''
+
+
+def write_tally_h5m(tally_list, filename):
+    # tally list contains elements of the form ('photon', ('current', ['Surface:4', 'Volume:1']))
     # loop over list
     for tally in tally_list:
-        name_flux = tally[1]
-        
-        particle_name = (name_flux.split(':'))[1].split('/')[0]
-        if not is_valid(particle_name):
-            print "particle: "+particle_name+" in tally "+name_flux+" in volume "+str(tally[0])
-            exit
-        tally_type = (name_flux.split('/'))[1].split('\n')[0]
-        tally_name = particle_name[0]+particle_name[1].upper()+tally_type[0].upper()+tally_type[1].upper()+str(tally[0])
-        print particle_name, tally_type, tally_name, tally[0], str(tally[0])
-        new_tally = Tally(tally_type,particle_name,tally[0],"Volume",str(tally[0]),tally_name,0.0,1.0)
-        new_tally.write_hdf5(filename,"/tally")
+        particle_name = tally[0][0].upper() + tally[0][1:len(tally[0])]
+        tally_type = tally[1][0]
+        for k in range(len(tally[1][1])):
+            tally_object = tally[1][1][k].split(':')[0]
+            object_id = tally[1][1][k].split(':')[1]
+            tally_name = particle_name[
+                0:2].upper() + tally_type[0: 7 - len(object_id)] + str(object_id)
+            new_tally = Tally(tally_type, particle_name,
+                              int(object_id), tally_object, str(object_id), tally_name, 0.0, 1.0)
+            new_tally.write_hdf5(filename, "/tally")
+
 """
 function to parse the script, adding options:
 defining 
 -f  : the .h5m file path
--d  : nuc_data path
--o  : name of the output h5m file "NAME.h5m"
+-d  : nuc_data.h5 'the PyNE material library' path
+-o  : name of the output h5m file "NAME.h5m", if not set the input .h5m file will be used
 """
+
+
 def parsing():
     parser = argparse.ArgumentParser()
     parser.add_argument(
```

Comment:
The `datafile` argument required to run `uwuw_preproc`, so it should should not be an option, but rather a positional argument. Change this by remove the -f:
```
parser.add_argument(action='store', dest='datafile', help='The path to the .h5m file')
````

The same is true of the -o flag.